schema/ksl-schema.kdl

```diff
@@ -0,0 +1,925 @@
+                                            min 1
+                                            type string
+                                        }
+                                    }
+                                    node allow-others about="""
+                                        Allow other properties
+
+                                        If present, allows other properties that
+                                        don't match this validator.
+                                        """ {
+                                        arg {
+                                            type boolean
+                                            default #true
+                                        }
+                                    }
+                                }
+                            }
+                            node arg about="""
+                                Defines an individual, ordered argument
+
+                                Each nth instance of this node will specify
+                                validations for the corresponding nth instance
+                                of the arg. Every specified `arg` is required,
+                                in the given order, unless marked as `optional`.
+                                """ {
+                                ref about-mixin
+                                ref value-validations
+                                repeatable
+                                children {
+                                    node optional about="""
+                                        Argument is not required
+
+                                        Specified `arg`s are required by
+                                        default.
+
+                                        `optional` only applies to *presence*:
+                                        an existing argument in an optional
+                                        `arg` \"slot\" that fails validation
+                                        will fail normally, even though it is
+                                        optional. As such, `optional` is only
+                                        really useful if it is on the last
+                                        `arg`, or is only followed by optional
+                                        `arg`s.
+                                        """ {
+                                        arg {
+                                            type boolean
+                                            default #true
+                                        }
+                                    }
+                                }
+                            }
+                            // TODO: add a feature that will let us specify that `args`
+                            // MUST be after any existing `arg` nodes in the current
+                            // scope. i.e. you can't do `node x { args; arg }`
```

Comment:
Could be a child node like `after` or `before` with a reference?

So in `args` it could be:
```kdl
node args {
    after super.arg
}
```

For example!